sol6-fallback-receive-split.md

```diff
@@ -0,0 +1,17 @@
+# Solidity 0.6.x features: Fallback functions split
+
+In versions of Solidity before 0.6.x, developers typically added a [fallback function](https://solidity.readthedocs.io/en/v0.5.15/contracts.html#fallback-function) as a function that a contract executed if no other functions matched the function called, or if a contract received ether, but no data.
+
+However, we realised that the dual-purpose of the function caused confusion to developers, and at worst, potential security problems.
+
+For example, a developer typically implements a fallback function, expecting only ether transfers to call it, but it is also called when a function is missing from a contract. Even worse, as this was expected behaviour, no error is reported.
```

Comment:
This sounds a little harsh to say that this is "even worse since no error was reported" as this was actually desired behaviour by many _proxy_ contracts that relied on this behaviour. E.g. the Delegate proxy pattern described and used across a large number of upgradable contracts.

Reply:
The example should probably even come from one of these "proxy" contracts, maybe the [Zeppelin one](https://github.com/OpenZeppelin/openzeppelin-sdk/blob/master/packages/lib/contracts/upgradeability/Proxy.sol)? Also [Colony](https://colony.io/dev/docs/colonynetwork/docs-upgrade-design), [Gnosis](https://blog.gnosis.pm/solidity-delegateproxy-contracts-e09957d0f201), Aragorn and many others use it.

Reply:
Then we should clarify that sentence. The crucial part is "expecting only ether transfers". If the fallback function is used in a proxy for both receiving ether and forwarding calls, that is perfectly fine. The problematic use is when calls are not forwarded.